server.go

```diff
@@ -1,28 +1,88 @@
 package main
 
 import (
+	"time"
+	"strconv"
+	"encoding/json"
+	"database/sql"
+	_ "github.com/go-sql-driver/mysql"
 	"github.com/gorilla/mux"
 	"github.com/gorilla/websocket"
 	"github.com/urfave/negroni"
 	"net/http"
 )
 
 var upgrader = websocket.Upgrader{}
+var DBUSER = env.Getenv("DBUSER")
+var DBPASS = env.Getenv("DBPASS")
+var DBNAME = env.Getenv("DBNAME")
+
+//Struct to hold Player data
+type User struct{
+	Name string
+	Time string
+}
+
+//function to get database connection
+func getConnection() (conn *sql.DB){
+	conn, err := sql.Open("mysql",DBUSER+":"+DBPASS+"@/"+DBNAME)
+	if err != nil{
+		panic(err)
+	}
+	return 
+}
+
+//function to get top players to display on leaderboard 
+func getPlayers(conn *sql.DB, grid_size int, difficulty int) (users []User){
+	query := "SELECT Name, Time FROM Leaderboard WHERE Sudoku_size = ? AND Difficulty = ?"
```

Comment:
Query should be like - 
`query := "SELECT Name, Time FROM Leaderboard WHERE Sudoku_size = ? AND Difficulty = ? ORDER BY Time LIMIT 5"`
As we want top 5 players according to their time

Reply:
Done